test/lit.cfg

```diff
@@ -311,6 +311,40 @@ if test_sdk_overlay_dir is not None:
 test_options = os.environ.get('SWIFT_TEST_OPTIONS')
 config.swift_test_options = ' ' + test_options if test_options else ''
 
+def get_clang_library_dir():
+    llvm_lib_clang = os.path.join(config.llvm_libs_dir, 'clang')
+    if not os.path.exists(llvm_lib_clang):
+        return ''
+
+    versions = os.listdir(llvm_lib_clang)
+    if not len(versions):
+        return ''
+
+    version = max(versions)
+    return os.path.join(llvm_lib_clang, version, 'lib',
+                        platform.system().lower())
+
+config.clang_library_dir = get_clang_library_dir()
+
+if os.path.exists(config.clang_library_dir):
+    config.compiler_rt_libs = [lib for lib in
+                               os.listdir(config.clang_library_dir)
+                               if lib.startswith('libclang_rt.')]
+else:
+    config.compiler_rt_libs = []
+
+def check_compiler_rt_feature(name):
+    for lib in config.compiler_rt_libs:
+        if lib.startswith('libclang_rt.' + name + '_'):
+            config.available_features.add(name)
+            return
+
+check_compiler_rt_feature('profile')
+check_compiler_rt_feature('asan')
```

Comment:
IIRC, we're already using the `asan` feature to mean "running under ASan", not "ASan available". Please pick a different name for one of these.